utilix/rundb.py

```diff
@@ -387,34 +388,64 @@ def delete_mc_document(self, document):
-        os.makedirs(save_dir, exist_ok=True)
-        write_to = os.path.join(save_dir, filename)
-        if os.path.exists(write_to) and not force:
-            logger.debug(f"{filename} already exists at {write_to} and the 'force' flag is not set.")
-        else:
-            logger.debug(f"Downloading {filename} from gridfs...")
-            response = self._get(url)
-            with open(write_to, 'wb') as f:
-                f.write(response.content)
-            logger.debug(f'DONE. {filename} downloaded to {write_to}')
-        return write_to
-
-    def load_file(self, filename, save_dir=None, force=False):
-        if save_dir is None:
-            save_dir = os.path.join(os.environ.get("HOME"), '.gridfs_cache')
-        path = self.download_file(filename, save_dir=save_dir, force=force)
-        return io.read_file(path)
-
-    def upload_file(self, filepath):
+    # def download_file(self, filename, save_dir='./', force=False):
+    #     """Downloads file from GridFS"""
+    #     url = f'/files/{filename}'
+    #     os.makedirs(save_dir, exist_ok=True)
+    #     write_to = os.path.join(save_dir, filename)
+    #     if os.path.exists(write_to) and not force:
+    #         logger.debug(f"{filename} already exists at {write_to} and the 'force' flag is not set.")
+    #     else:
+    #         logger.debug(f"Downloading {filename} from gridfs...")
+    #         response = self._get(url)
+    #         with open(write_to, 'wb') as f:
+    #             f.write(response.content)
+    #         logger.debug(f'DONE. {filename} downloaded to {write_to}')
+    #     return write_to
+    #
+    # def load_file(self, filename, save_dir=None, force=False):
+    #     if save_dir is None:
+    #         save_dir = os.path.join(os.environ.get("HOME"), '.gridfs_cache')
+    #     path = self.download_file(filename, save_dir=save_dir, force=force)
+    #     return io.read_file(path)
+    #
+    # def upload_file(self, filepath):
+    #     with open(filepath, 'rb') as f:
+    #         fb = f.read()
+    #     filename = os.path.basename(filepath)
+    #     url = f'/files/{filename}'
+    #     return self._post(url, data=fb)
+
+    def count_documents(self, query: dict)->int:
+        """Perform collection.count_documents on the fs.files-collection using the query"""
+        """<URL MAGIC>"""
+        response = json.loads(self._get(url).text)
+        return response.get('results', 0)
+
+    def file_find(self, projection:dict)->dict:
+        """Perform collection.find(projection=projection) on the fs.files-collection"""
+        """<URL MAGIC>"""
+        response = json.loads(self._get(url).text)
+        return response.get('results', {})
+
+    def get_last_version(self, query: dict):
+        """Perform grid_fs.get_last_version(**query) on gridfs"""
+        """<URL MAGIC>"""
+        response = self._get(url)
+        return response.get('results', {})
+
+    def upload_file(self, filepath, metadata: dict):
```

Comment:
@ershockley you right, and the `added` field is indeed of no added value. I had envisaged that more elaborate schemes could be implemented, like versions, for CMT. However, this turned out to be unnecessary. I think this question is related to the question you ask here:
https://github.com/XENONnT/utilix/pull/38#discussion_r600053766

Let's indeed change it to only use the filename since that would make it indeed more simple from the API side.